### PR TITLE
Use range loop (if possible)

### PR DESCRIPTION
**Pull request by ShFil119**
Should behave as original code, but without magic numbers and looks a bit simpler.

// not sure if formatting will satisfy you :(

### CODE REVIEW

**Comment by JosJuice** [comment]
* Please don't touch the code in the Externals directory.
* There should be no space before `&`. (I'll run the lint bot for this PR so it can point out the incorrect formatting)
* Don't use the m_ prefix for local variables.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by ShFil119** [comment]
Should be ok now.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/DSP/DSPCore.cpp [hunk 1]
```diff
@@ -147,8 +147,8 @@ bool DSPCore_Init(const DSPInitOptions& opts)
 
   std::fill(std::begin(g_dsp.reg_stack_ptr), std::end(g_dsp.reg_stack_ptr), 0);
 
-  for (size_t i = 0; i < ArraySize(g_dsp.reg_stack); i++)
-    std::fill(std::begin(g_dsp.reg_stack[i]), std::end(g_dsp.reg_stack[i]), 0);
+  for (auto& i : g_dsp.reg_stack)
+    std::fill(std::begin(i), std::end(i), 0);
```
Comment (AdmiralCurtiss) [inline Source/Core/Core/DSP/DSPCore.cpp]: I dunno about using `i` for a loop value. Granted, in this case I'm not sure what a better name would be, but that just gives the wrong impression when glancing over the code.

Comment (BhaaLseN) [reply Source/Core/Core/DSP/DSPCore.cpp]: That'd be the only shortcomings I see with this PR - most of those single-letter iteration variables could use a better name. Or least don't call them `i` which is the most likely one to be used in a plain for loop as counter.

**Comment by ShFil119** [comment]
Renamed a bit.

**Comment by Tilka** [comment]
From the style guide:

> Do not use the auto keyword everywhere. While it's nice that the type can be determined by the compiler, it cannot be resolved at 'readtime' by the developer as easily. Use auto only in cases where it is obvious what the type being assigned is (note: 'obvious' means not having to open other files or reading the header file). Some situations where it is appropriate to use auto is when iterating over a std::map container in a foreach loop, or to shorten the length of container iterator variable declarations.

File: Source/Core/InputCommon/GCAdapter.cpp [hunk 2]
```diff
@@ -222,11 +222,8 @@ static void Setup()
   libusb_device** list;
   ssize_t cnt = libusb_get_device_list(s_libusb_context, &list);
 
-  for (int i = 0; i < SerialInterface::MAX_SI_CHANNELS; i++)
-  {
-    s_controller_type[i] = ControllerTypes::CONTROLLER_NONE;
-    s_controller_rumble[i] = 0;
-  }
+  std::fill(s_controller_type.begin(), s_controller_type.end(), ControllerTypes::CONTROLLER_NONE);
+  std::fill(s_controller_rumble.begin(), s_controller_rumble.end(), 0);
```
Comment (lioncash) [inline Source/Core/InputCommon/GCAdapter.cpp]:
```suggestion
  s_controller_type.fill(ControllerTypes::CONTROLLER_NONE);
  s_controller_rumble.fill(0);
```

File: Source/Core/InputCommon/GCAdapter.cpp [hunk 3]
```diff
@@ -370,8 +367,7 @@ static void Reset()
     s_adapter_output_thread.join();
   }
 
-  for (int i = 0; i < SerialInterface::MAX_SI_CHANNELS; i++)
-    s_controller_type[i] = ControllerTypes::CONTROLLER_NONE;
+  std::fill(s_controller_type.begin(), s_controller_type.end(), ControllerTypes::CONTROLLER_NONE);
```
Comment (lioncash) [inline Source/Core/InputCommon/GCAdapter.cpp]:
```suggestion
  s_controller_type.fill(ControllerTypes::CONTROLLER_NONE)
```

File: Source/Core/DolphinQt/RenderWidget.cpp [hunk 4]
```diff
@@ -333,6 +333,6 @@ void RenderWidget::SetImGuiKeyMap()
                                    {ImGuiKey_Y, Qt::Key_Y},
                                    {ImGuiKey_Z, Qt::Key_Z}};
   auto lock = g_renderer->GetImGuiLock();
-  for (size_t i = 0; i < ArraySize(key_map); i++)
-    ImGui::GetIO().KeyMap[key_map[i][0]] = (key_map[i][1] & 0x1FF);
+  for (auto [imGuiKey, qtKey] : key_map)
```
Comment (lioncash) [inline Source/Core/DolphinQt/RenderWidget.cpp]:
```suggestion
  for (auto [imgui_key, qt_key] : key_map)
```

**Comment by ShFil119** [comment]
Small up.

File: Source/Core/InputCommon/GCAdapter.cpp [hunk 5]
```diff
@@ -370,8 +367,7 @@ static void Reset()
     s_adapter_output_thread.join();
   }
 
-  for (int i = 0; i < SerialInterface::MAX_SI_CHANNELS; i++)
-    s_controller_type[i] = ControllerTypes::CONTROLLER_NONE;
+  s_controller_type.fill(ControllerTypes::CONTROLLER_NONE)
```
Comment (BhaaLseN) [inline Source/Core/InputCommon/GCAdapter.cpp]: You should still add the missing `;` here so it compiles.

Comment (ShFil119) [reply Source/Core/InputCommon/GCAdapter.cpp]: Ah, I've messed with rebasing.

**Comment by ShFil119** [comment]
Rebased.

**Comment by ShFil119** [comment]
Fixed linter error.

**Comment by jordan-woyak** [comment]
Can we get another rebase, please?

**Comment by ShFil119** [comment]
@jordan-woyak, should be ok now.

**Comment by ShFil119** [comment]
Thx for help with formatting.

**Comment by jordan-woyak** [comment]
No problem. I configure my editor to clang-format on save so I can't forget.

**Pull request merged**